src/client/Microsoft.Identity.Client/IntuneAppProtectionPolicyRequiredException.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// This exception is thrown when Intune requires App protection policy.
+    /// Properties in it can be used by app to obtain the required enrollmentID from MAM SDK
+    /// https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Exceptions/#IntuneAppProtectionPolicyRequiredException
+    /// </summary>
+    public class IntuneAppProtectionPolicyRequiredException : MsalServiceException
```

Comment:
Should this have Intune in the name or be more generic, `AppProtectionPolicyRequiredException`? Will this behavior apply to any other MAMs?

Reply:
Good point. This is consistent with Android SDK. What do you think @bgavrilMS @trwalke @maliksahil ?

Reply:
+1 on consistency